### ISSUE TITLE
[ENH] Implement TSelect as a BaseChannelSelector for multivariate time series

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#3389]
### Describe the feature or idea you want to propose

TSelect 
https://link.springer.com/article/10.1007/s10618-025-01132-4
is a channel selection algorithm that is classifier-agnostic. It would be great to have it in aeon.

Algorithm outline:
**1. Per-channel relevance scoring**
For each channel independently:

extract 6 summary features: min, max, mean, variance, skewness, kurtosis
split the training set into:
learning set: max(100, n_cases // 4)
validation set: remainder
train a simple classifier on the learning set
compute validation ROCAUC

Store:
AUC score per channel
validation predicted probabilities per channel

**2. Irrelevant channel removal**

remove channels with ROCAUC < 0.5
from the remaining channels, remove the worst alpha%

**3. Redundant channel removal**

compare remaining channels using the predictions from step 1
for each class:
rank examples by predicted probability
compute pairwise Spearman rank correlation
average across classes
cluster channels using the threshold corr_threshold
keep the highest-ROCAUC channel from each cluster

Final output:

selected channel indices in channels_selected_

### Describe your proposed solution

TSelect should inherit from BaseChannelSelector and output a subset of the original channels.

```python
class TSelect(BaseChannelSelector):
    def __init__(
        self,
        alpha=0.4,
        corr_threshold=0.7,
        estimator=None,
        random_state=None,
    ):
```

Suggested parameters:

alpha : float, default=0.4
Proportion of the remaining channels to remove after the relevance stage.
corr_threshold : float, default=0.7
Correlation threshold used in redundancy pruning.
estimator : object or None, default=None
Simple per-channel estimator used in the relevance stage. If None, use LogisticRegression.
random_state : int or None, default=None
Random seed for reproducibility.

Possible learned attributes:

channels_selected_
channel_scores_
channel_correlations_
clusters_

binary and multiclass classification support
use sklearn.linear_model.LogisticRegression by default
use roc_auc_score
use a simple threshold-based clustering / greedy grouping first
refine later if needed to better match the paper’s exact clustering lo

**Comment by TonyBagnall** [comment]
author version is here https://github.com/ML-KULeuven/TSelect

**Comment by LorenNuyts** [comment]
Thanks for working on this! Happy to provide more detail to help with the implementation.

**Implementation location**

The main implementation is in `src/tselect/channel_selectors/tselect.py`. The class contains some leftover code from experiments, but if you only expose the parameters described in your issue the rest will either be unused or fall back to sensible defaults.

**Parameter mapping**

| Your proposed parameter | Corresponding `TSelect.__init__` parameter | Notes |
|---|---|---|
| `alpha` | `irrelevant_percentage_to_keep` | `irrelevant_percentage_to_keep = 1 - alpha`, so the default `alpha=0.4` corresponds to `irrelevant_percentage_to_keep=0.6` |
| `corr_threshold` | `redundant_correlation_threshold` | Direct match, default `0.7` |
| `estimator` | *(not currently a parameter)* | See note below |
| `random_state` | `random_state` | Direct match |

There are also two boolean flags worth exposing if you want full control:
- `irrelevant_selector`: whether to run the irrelevance filter at all (default `True`)
- `redundant_selector`: whether to run the redundancy filter at all (default `True`)

And one additional parameter:
- `irrelevant_hard_threshold` : the hard lower bound on ROCAUC below which a channel is always dropped, regardless of `alpha` (default `0.5`). This is applied before the top-`(1-alpha)%` percentage cut.

**Validation (learning) set size**

The `validation_size` parameter overrides the default split size. If left as `None`, the split is computed as follows: the training portion is `max(100, round(0.25 * n_instances))` instances, and the rest goes to the validation set. This means for small datasets the model is given more training data; for larger datasets (~400+ instances) roughly 75% is used for validation/AUC scoring.

**Feature extraction**

By default, six summary statistics are extracted per channel using a single pass: min, max, mean, variance, skewness, and kurtosis, which matches what you described in the issue. In the original code these come from `SinglePassStatistics` in TSFuse, using Cython.

**Custom estimator**

The current implementation hardcodes `LogisticRegression` and does not expose the classifier as a parameter. Your proposal to add an `estimator` parameter is a reasonable extension. It would just need to be passed through to the `train_models` method in place of the hardcoded `LogisticRegression(random_state=...)` call.

**Output attribute mapping**

| Your proposed attribute | Source in `TSelect` |
|---|---|
| `channels_selected_` | `self.selected_channels` |
| `channel_scores_` | `self.auc_col` (dict mapping channel → ROCAUC) |
| `channel_correlations_` | `self.rank_correlation` |
| `clusters_` | `self.clusters` |

**Edge case: all channels fail the hard threshold**

If no channels pass the `irrelevant_hard_threshold` check, the algorithm falls back to keeping all channels rather than returning an empty selection, and emits a warning. This is intentional — it ensures the downstream classifier always receives some input.

**The `config` parameter**

The `Config` object was introduced to make hyperparameter experimentation easier and bundles all of the above into one object. For the aeon integration you can safely ignore it and just use the individual keyword arguments directly.
